src/cpp/proxy/engine.cpp

```diff
@@ -776,13 +777,13 @@ private slots:
 		}
 	}
 
-	void ps_finished()
+	void ps_finished(ProxySession *ps)
 	{
-		ProxySession *ps = (ProxySession *)sender();
-
 		ProxyItem *i = proxyItemsBySession.value(ps);
 		assert(i);
 
+		finishedConnection.erase(ps);
```

Comment:
Need to delete from the other maps too.

Reply:
thanks for catching this one.
I also simplified another conn map erase `d->finishedConnections.erase(req);`